include/rest/api_client.hpp

```diff
@@ -130,8 +129,10 @@ namespace disccord
 
                     pplx::task<void> request_empty(route_info& route, const pplx::cancellation_token& token = pplx::cancellation_token::none())
                     {
-                        set_content_length(0);
-                        return request_internal(route, token).then([](web::http::http_response response){});
+                        disccord::api::request_info* info = new disccord::api::request_info();
+                        
+                        info->set_body("", "application/json");
```

Comment:
@FiniteReality  Not all methods using this will be a GET, for example, the create_reaction method I added here is a PUT. This is just here so it will trigger the call later in the request that will set the content-length. since you make a check to see if there is a body before setting it in the web::http::http_request.
In testing the create_reaction endpoint, here is response without setting this body:
```
HTTP/1.1 411 Length Required
CF-RAY: 3474c7876fab2a5b-SEA
Connection: keep-alive
Content-Type: text/html; charset=UTF-8
Date: Wed, 29 Mar 2017 18:09:18 GMT
Server: cloudflare-nginx
Set-Cookie: __cfduid=deb617dbe1e7b67c591c0ccd658ecafba1490810957; expires=Thu, 29-Mar-18 18:09:17 GMT; path=/; domain=.discordapp.com; HttpOnly
Transfer-Encoding: chunked
```

Here is the response with setting this body:
```
HTTP/1.1 204 NO CONTENT
Alt-Svc: clear
CF-RAY: 3474d2c68e5f2a85-SEA
Connection: keep-alive
Content-Type: text/html; charset=utf-8
Date: Wed, 29 Mar 2017 18:16:58 GMT
Server: cloudflare-nginx
Set-Cookie: __cfduid=d113e85e9595e8a7de6ede001017159051490811418; expires=Thu, 29-Mar-18 18:16:58 GMT; path=/; domain=.discordapp.com; HttpOnly
Strict-Transport-Security: max-age=31536000; includeSubDomains
Via: 1.1 google
X-RateLimit-Limit: 1
X-RateLimit-Remaining: 0
X-RateLimit-Reset: 1490811419

```
So it seems we have to do something here, can be approached multiple ways.